test/lit/passes/global-effects.wast

```diff
@@ -423,7 +423,7 @@
   ;; WITHOUT-NEXT:  )
   ;; WITHOUT-NEXT: )
   ;; INCLUDE:      (func $call-throw-or-unreachable-and-catch (type $2) (param $x i32)
-  ;; INCLUDE-NEXT:  (try $try
+  ;; INCLUDE-NEXT:  (try
```

Comment:
The amount of changes in this PR is very large. Is it worth trying to reduce diffs like this in preparatory PRs?

If not, can you detail the reasons for the various categories of differences? Some were puzzling to me. I can write specific questions if you prefer.

Reply:
The categories I'm seeing are:

1. Elem and data segments get slightly different generated names.
2. Loops no longer get generated names if there are no branches.
3. Source mappings are no longer automatically applied to epilogues.
4. Empty then and else clauses are parsed as nops instead of empty blocks.
5. Some tests are fixed to properly use strings in export clauses.
6. Parse error tests are updated.
7. Exports are parsed in a slightly different order.
8. Some tests are updated to respect import ordering constraints.
9. Tests of pops inside of blocks are deleted since they no longer parse. @aheejin thought this would be ok.
10. Old pre-1.0 spec tests that use nonstandard syntax are removed.
11. The tests for parsing Poppy IR directly are removed.

and of course the big one:

12. Try expressions are parsed with wrapper blocks and their labels are replaced with \_\_delegate\_\_ labels.

I'll think about whether the fixing the last one would be feasible since it accounts for the vast majority of hard-to-understand diff-verbosity. I'll also try to pick some low-hanging fruit for the other ones, but I think at some point minimizing the diff more will no longer be worth the extra complexity.

Reply:
After #6505, the number of lines added in the diff went down by about 2/3.